src/v3/src/components/PasswordWithConfirmation/PasswordWithConfirmation.tsx

```diff
@@ -0,0 +1,170 @@
+
+enum PasswordType {
+  NEW,
+  CONFIRM,
+}
+
+const PasswordWithConfirmation: UISchemaElementComponent<{
+  uischema: PasswordWithConfirmationElement
+}> = ({ uischema }) => {
+  const { input } = uischema.options;
+  const {
+    name: newPwName,
+    label: newPwLabel,
+    options: {
+      inputMeta: {
+        // @ts-ignore expose type from auth-js
+        messages = {},
+      },
+      attributes,
+    },
+  } = input;
+
+  const { dataSchemaRef } = useWidgetContext();
+  const value = useValue(input);
+  const onChangeHandler = useOnChange(input);
+  // Must use this flag to determine which field contains error
+  const [hasNewPwError, setHasNewPwError] = useState<boolean>(false);
+  const [isNewPwTouched, setIsNewPwTouched] = useState<boolean>(false);
+  const [isTouched, setIsTouched] = useState<boolean>(false);
+  const [confirmPassword, setConfirmPassword] = useState<string>();
+  const newPasswordError = hasNewPwError && messages?.value?.[0]
+    && getMessage(messages.value[0]);
+  const confirmPasswordError = !hasNewPwError
+    ? (messages?.value?.[0] && getMessage(messages.value[0]))
+    : (messages?.value?.[1] && getMessage(messages.value[1]));
+
+  // Overrides default validate function for password field
+  const validate = useCallback((data: FormBag['data']) => {
+    const newPw = data[input.name];
+    const errorMessages: Partial<IdxMessage>[] = [];
+    if (!newPw) {
+      setIsNewPwTouched(false);
+      setHasNewPwError(true);
+      errorMessages.push({ i18n: { key: 'model.validation.field.blank' } });
+    } else {
+      setHasNewPwError(false);
+    }
+
+    if (!confirmPassword) {
+      setIsTouched(false);
+      errorMessages.push({ i18n: { key: 'model.validation.field.blank' } });
+    } else if (confirmPassword !== newPw) {
+      setIsTouched(false);
+      errorMessages.push({ i18n: { key: 'password.error.match' } });
+    }
+
+    return errorMessages.length ? errorMessages : undefined;
+  }, [confirmPassword, input.name, setHasNewPwError, setIsTouched, setIsNewPwTouched]);
+
+  const handlePasswordChange = (e: ChangeEvent<HTMLInputElement>, type: PasswordType) => {
+    if (type === PasswordType.NEW) {
+      setIsNewPwTouched(true);
+      onChangeHandler(e.currentTarget.value);
+    } else {
+      setIsTouched(true);
+      setConfirmPassword(e.currentTarget.value);
+    }
+  };
+
+  const handlePasswordBlur = (type: PasswordType) => {
+    if (type === PasswordType.NEW) {
+      setIsNewPwTouched(true);
+    } else {
+      setIsTouched(true);
+    }
+  };
+
+  useEffect(() => {
+    // update validate function to prevent submission w/o confirm pw data
+    // when server messages are set, we must reset the validate function
+    dataSchemaRef.current![input.name].validate = validate;
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [validate, newPasswordError, confirmPasswordError, input.name]);
+
+  return (
+    <Box>
+      <Box marginBottom={4}>
+        <PasswordInput
```

Comment:
You can have two uischema states, one for password, and one for confirmPassword, and pass each uischema to its InputPassword component. 

Also make InputPassword can accept `value` props to override `const value = useValue(uischema)`, so you can pass state down to that component.

Reply:
Other than the label and name, what benefit do you get out of creating a separate schema element that is not from the transaction, that will only be used in this component?

Reply:
DRY

Reply:
if the component is hard to be reused, which most probably means that component is not well organized

Reply:
Are you saying to do something like this? 

```
export interface PasswordWithConfirmationElement extends UISchemaElement {
  type: 'PasswordWithConfirmation';
  options: {
    password: FieldElement,
    confirmPassword: FieldElement,
  };
}
```